react/README.md

```diff
@@ -34,12 +34,28 @@ For example, the following works:
 <MaskedTextInput
   mask={['(', /[1-9]/, /\d/, /\d/, ')', ' ', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/]}
   className="form-control"
-  placeholder="Enter zip code"
+  placeholder="Enter a phone number"
   guide={false}
   id="my-input-id"
 />
 ```
 
+You can additionally use your own custom input component by passing in the prop `inputElement`.
```

Comment:
They would need to pass in an input element. Problem with that is on first render you have nothing. It needs to a callback.

The only solution I've come up with is not to include `react-dom` in `text-mask` but to add a `getNode` prop that takes a function. Then if the user has defined this they would need to add their own ref to `text-mask` and then additionally return `ReactDOM.findDOMNode(this.myRef)` in the `getNode` callback.

```
<TextMask 
  customInput={FormControl}
  ref={(comp) => this.myTextMaskRef = comp}
  getNode={() => ReactDOM.findDOMNode(this.myTextMaskRef)}
/>
```

This is less than up front but partially the only option if there are deeply nested React Components inside the Component that is passed into `customInput`.

I need to think/look into this a little more before I make a final design choice.

Another idea I was toying with is if someone passes a child in then clone that and attach the ref. But now we are at the same issue.